lib/shoryuken/options.rb

```diff
@@ -47,6 +48,16 @@ def add_queue(queue, weight, group)
         end
       end
 
+      def group_for_queue(queue)
+        groups.each do |group|
+          # note, there is a bug here where two queues with the same name
+          # will choose the first queue and group
+          if group[1][:queues].include?(queue)
+            return group.first # the group name
+          end
```

Comment:
@phstc Going from the queue name string to the group delay configuration here. Is there any existing validation that a queue name must be unique across all groups?

Reply:
@lucas-ibotta the same queue can be in multiple groups. We can't get the delay based on the queue name.

What I think we could do, is to change the `PollingStrategy` to receive a second argument `delay = nil` (default to `nil` to avoid breaking changes) then [here](https://github.com/phstc/shoryuken/blob/352b20642b4b8a123b821228231ed5a5c618a9cc/lib/shoryuken/launcher.rb#L75) pass it as the second argument, similar to [this](https://github.com/phstc/shoryuken/blob/352b20642b4b8a123b821228231ed5a5c618a9cc/lib/shoryuken/options.rb#L78-L79).

```ruby
 def create_managers
  Shoryuken.groups.map do |group, options|
    Shoryuken::Manager.new(
      Shoryuken::Fetcher.new(group),
      Shoryuken.polling_strategy(group).new(options[:queues], Shoryuken.delay(group)),
      options[:concurrency],
      executor
    )
  end
end
```


Then in `BaseStrategy#queue` we:

```ruby
def delay
  @delay || Shoryuken.options[:delay].to_f
end
```

WDYT?